src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -22,9 +22,8 @@ abstract class KernelGenerator extends Transformer[DenseVector[Double], DenseVec
   /* Cached KernelBlockBlock (used for training), and corresponding
    * blockIdx
    */
-  var cachedBlockBlock:Option[(DenseMatrix[Double], Seq[Int])] = None
 
-  def apply(testMat: RDD[DenseVector[Double]], blockIdxs: Seq[Int], same: Boolean = false, cache: Boolean = false): RDD[DenseVector[Double]]
+  def apply(testMat: RDD[DenseVector[Double]], blockIdxs: Seq[Int], same: Boolean = false): RDD[DenseVector[Double]]
```

Comment:
Let's be careful about adding more `FunctionNode`s - if anything I'd like to figure out how to get rid of them (or find a way to make them first class citizens). 

I think the problem is that `Transformers` and `Estimators` are typed at the item level, and we are conceptualizing `KernelGenerator` at the collection level.

One simple fix, potentially, is to have `KernelMatrix extends RDD[DenseVector[Double]]` and have it also have the block generator interface we talked about. 

Then, if we have:
`val kernelRidgePipeline = Identity andThen (new GaussianKernelGenerator(), trainData) andThen (new LinearSystemSolver(), trainData, trainLabels)`

The types are as follows:

```scala
GaussianKernelGenerator: Estimator[DenseVector[Double], DenseVector[Double]] {
  def fit(in: RDD[DenseVector[Double]]): Kernelizer = ???
}

class Kernelizer: Transformer[DenseVector[Double],DenseVector[Double]] {
  override def apply(in: RDD[DenseVector[Double]]): KernelMatrix = ??? //Yields the whole kernel matrix
  def apply(in: DenseVector[Double]): DenseVector[Double] = ??? //Do this for a single point.
}

class KernelMatrix extends RDD[DenseVector[Double]] {
  def getIndices(x: Seq[Int]): Block = ??? //Block generating logic
}

class LinearSystemSolver extends LabelEstimator[DenseVector[Double],DenseVector[Double],DenseVector[Double]] {
  def fit(in: RDD[DenseVector[Double]): LinearMapper = {
    in match {
     case x: KernelMatrix => //do something with x.getIndices here and generate a LinearMapper
     case _ => //Eh, maybe we don't implement this for now?
   }
  }
}
```

Does this look sane or am I missing something @Vaishaal @shivaram ?